### PR TITLE
CC-16229 CC-16234 configurable_bundle_endpoints

### PR DESCRIPTION
**Pull request by pyvovarov-s**
CC-16229 CC-16234 configurable_bundle_endpoints

### CODE REVIEW

File: tests/api/b2b/glue/configurable_bundle_endpoints/configurable_bundle/positive.robot [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    And Response reason should be:    OK
+    And Response header parameter should be:    Content-Type    ${default_header_content_type}
+    And Response body parameter should be:    [data][type]    carts
+    And Response body parameter should be:    [data][id]    ${cart_id}
+    And Response body parameter should be:    [data][attributes][priceMode]    ${gross_mode}
+    And Response body parameter should be:    [data][attributes][currency]    ${currency_code_eur}
+    And Response body parameter should be:    [data][attributes][store]    ${store_de}
+    And Response body parameter should be:    [included][0][type]    items
+    And Response body parameter should be:    [included][0][attributes][sku]    ${configurable_bundle_first_slot_item_sku}
+    And Response body parameter should be:    [included][0][attributes][configuredBundle][quantity]   12
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/configurable_bundle_endpoints/configurable_bundle/positive.robot:137]: can we also check that grandtotal increased after qty increase?

File: tests/api/b2b/glue/configurable_bundle_endpoints/configurable_bundle/positive.robot [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...  AND    I set Headers:    Authorization=${token}
+    ...  AND    I send a POST request:    /carts    {"data": {"type": "carts","attributes": {"priceMode": "${gross_mode}","currency": "${currency_code_eur}","store": "${store_de}","name": "${test_cart_name}-${random}"}}}
+    ...  AND    Save value to a variable:    [data][id]    cart_id
+    ...  AND    Response status code should be:    201
+    ...  AND    I send a POST request:    /carts/${cart_id}/configured-bundles?include=items    {"data": {"type": "configured-bundles","attributes": {"quantity": "${configured_bundle_quantity}","templateUuid": "${configurable_bundle_template_id}","items": [{"sku": "${configurable_bundle_first_slot_item_sku}","quantity": 2,"slotUuid": "${configurable_bundle_first_slot_uuid}"}]}}}
+    ...  AND    Save value to a variable:    [included][0][attributes][configuredBundle][groupKey]    bundle_group_key
+    ...  AND    Response status code should be:    201
+    When I send a DELETE request:    /carts/${cart_id}/configured-bundles/${bundle_group_key}
+    Then Response status code should be:    204
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/configurable_bundle_endpoints/configurable_bundle/positive.robot:151]: can we get the cart and see that there is no bindle in it and grandtotal was decreased?

File: tests/api/b2b/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+Add_configured_bundle_item_to_cart_with_invalid_properties
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Content-Type=${default_header_content_type}    Authorization=${token}
+    ...    AND    I send a POST request:    /carts    {"data":{"type":"carts","attributes":{"priceMode":"${gross_mode}","currency":"${currency_code_eur}","store":"${store_de}","name":"Cart-${random}"}}}
+    ...    AND    Save value to a variable:    [data][id]    cart_id
+    When I send a POST request:    /carts/${cart_id}/configured-bundles    {"data": {"type": "configured-bundles","attributes": {"quantity": "","templateUuid": "","items": [{"sku": "","quantity": 2,"slotUuid": ""}]}}}
+    Then Response status code should be:    422
+    And Response reason should be:    Unprocessable Content
+    And Response should return error code:    4003
+    And Response should return error message:    The quantity of the configured bundle should be more than zero.
```
[the remark this answers is not in the document]
Comment (pyvovarov-s) [reply tests/api/b2b/glue/configurable_bundle_endpoints/configurable_bundle/negative.robot:110]: That's a tricky question cause we don't have any documentation for that functionality yet

File: tests/api/b2b/glue/configurable_bundle_endpoints/configurable_bundle/positive.robot [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    And Response should contain the array of a certain size:    [data]    2
+    And Each array element of array in response should contain property with value:    [data]    type    configurable-bundle-templates
+    And Each Array Element Of Array In Response Should Contain Property:    [data]    id
+    And Each Array Element Of Array In Response Should Contain Property:    [data]    attributes
+    And Each Array Element Of Array In Response Should Contain Property:    [data]    links
+    And Each array element of array in response should contain nested property:    [data]    [attributes]    name
+    And Each array element of array in response should contain nested property:    [data]    [links]    self
+    Response body has correct self link
+
+Get_configuraable_bundle_templates_including_configuraable_bundle_template_slots
```
[the remark this answers is not in the document]
Comment (pyvovarov-s) [reply tests/api/b2b/glue/configurable_bundle_endpoints/configurable_bundle/positive.robot:25]: added bug reported BUG CC-16634

File: tests/api/b2b/glue/configurable_bundle_endpoints/configurable_bundle/positive.robot [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    ...  AND    Response status code should be:    201
+    ...  AND    I send a POST request:    /carts/${cart_id}/configured-bundles?include=items    {"data": {"type": "configured-bundles","attributes": {"quantity": "${configured_bundle_quantity}","templateUuid": "${configurable_bundle_template_id}","items": [{"sku": "${configurable_bundle_first_slot_item_sku}","quantity": 2,"slotUuid": "${configurable_bundle_first_slot_uuid}"}]}}}
+    ...  AND    Save value to a variable:    [included][0][attributes][configuredBundle][groupKey]    bundle_group_key
+    ...  AND    Response status code should be:    201
+    When I send a DELETE request:    /carts/${cart_id}/configured-bundles/${bundle_group_key}
+    Then Response status code should be:    204
+    And Response reason should be:    No Content
+    And I send a GET request:    /carts/${cart_id}
+    And Response body parameter should be:    [data][attributes][totals][priceToPay]    0
+    And Response body parameter should be:    [data][attributes][totals][grandTotal]    0
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/configurable_bundle_endpoints/configurable_bundle/positive.robot:186]: should not we add teardown to delete cart?

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**